src/App.tsx

```diff
@@ -15,47 +16,123 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+interface ReorderOptions {
+  sortType: SortType,
+  isReversed: boolean,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReverse] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const hendleAlphabeticallySort = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const hendleLengthSort = () => {
+    setSortType(SortType.LENGTH);
+  };
```

Comment:
fix it in all the cases

```suggestion
  const handleLengthSort = () => {
    setSortType(SortType.LENGTH);
  };
```